youtube_transcript_api/_api.py

```diff
@@ -48,15 +52,27 @@ def list_transcripts(cls, video_id, proxies=None):
         :type video_id: str
         :param proxies: a dictionary mapping of http and https proxies to be used for the network requests
         :type proxies: {'http': str, 'https': str} - http://docs.python-requests.org/en/master/user/advanced/#proxies
+        :param cookies: a string of the path to a text file containing youtube authorization cookies
+        :type cookies: str - cookies.txt
         :return: the list of available transcripts
         :rtype TranscriptList:
         """
         with requests.Session() as http_client:
+            if cookies:
+                try:
+                    cj = cookiejar.MozillaCookieJar()
+                    cj.load(cookies)
+                    http_client.cookies = cj
+                except IOError as e:
+                    print("Warning: Path for cookies file was not valid. Did not load any cookies")
```

Comment:
I do not think that we should this. This will swallow errors and just continue execution although an exception occurred. 
Did you implement the print to notify the CLI users? If so, I would prefer a custom exception in `_errors`. The way the CLI is implemented, subtypes of `CouldNotRetrieveTranscript` will always just print the error message to the user. That way CLI users get a printed error message while API users will get an exception which they can handle. I think an error should always kill the application unless it is handled specifically!

Reply:
My reasoning was that many videos don't need cookies so its possible the call (or in get transcripts, some of the calls) would still work. I can have it kill the app if you would like. However, do you wan the the custom exceptions I make inherit from `CouldNotRetrieveTranscript`?

Reply:
I am not sure if I am understanding you correctly. Those errors won't be raised if the video doesn't need a cookie, only when the cookies path is invalid. If you supply a path to a cookie you want the cookie to be used, I'd say, whether the video needs a cookie or not. If that path is invalid, it should fail.
And yes, `CouldNotRetrieveTranscript` is meant to be the base class for all exceptions by this module. Although the naming might not be optimal in this case...